pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,347 @@
+        ":BATT:TEST:SENS:AH:VFUL %g",
+        """Set the charging target voltage for battery tests.""",
+        validator=strict_range,
+        values=_VOLTAGE_RANGE
+    )
+
+    bt_charge_current_limit = Instrument.control(
+        ":BATT:TEST:SENS:AH:ILIM?",
+        ":BATT:TEST:SENS:AH:ILIM %g",
+        """Set the maximum charging current for battery tests.""",
+        validator=strict_range,
+        values=_CURRENT_RANGE_BT_BS
+    )
+
+    bt_termination_current = Instrument.control(
+        ":BATT:TEST:CURR:END?",
+        ":BATT:TEST:CURR:END %g",
+        """Set the termination current for battery charging and discharging.""",
+        validator=strict_range,
+        values=[0, 0.1]
+    )
+
+    bt_output_enabled = Instrument.control(
+        ":BATT:OUTP?",
+        ":BATT:OUTP %s",
+        """Control the output state.""",
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True
+    )
+
+    bt_test_control = Instrument.setting(
+        ":BATT:TEST:SENS:AH:EXEC %s",
+        """Control the output state.""",
+        validator=strict_discrete_set,
+        values={"START", "STOP", "PAUSE", "CONTINUE"}
+    )
+
+    bt_save_model_internal = Instrument.setting(
+        ":BATT:TEST:SENS:AH:GMOD:SAVE:INTE %d",
+        """Save the battery model to internal memory.""",
+        validator=strict_discrete_set,
+        values=_INTERNAL_MEMORY_SLOTS
+    )
+
+    bt_save_model_usb = Instrument.setting(
+        ":BATT:TEST:SENS:AH:GMOD:SAVE:USB \"%s\"",
+        """Save the battery model to usb memory."""
+    )
+
+    bt_buffer_data = Instrument.measurement(
+        ":BATT:DATA:DATA? \"VOLT, CURR, RES, AH, REL\"",
+        """Get the buffer in battery test mode and return its content as a pandas dataframe""",
+        separator=",",
+        get_process=lambda v: Keithley2281S._bt_parse_buffer(v)  # Does not work without lambda
+    )
+
+    @staticmethod
+    def _bt_parse_buffer(buffer_content):
+        if len(buffer_content) < 3:
+            return pd.DataFrame({'current': [], 'voltage': [], 'capacity': [], 'resistance': [], 'time': []})
+        data = np.array(buffer_content, dtype=np.float32)
+        return pd.DataFrame({'voltage': data[0::5], 'current': data[1::5],
+                             'resistance': data[2::5], 'capacity': data[3::5], 'time': data[4::5]})
+
+    # Battery simulation commands, only applicable in battery simulator mode
+
+    bs_select_model_internal = Instrument.setting(
+        ":BATT:MOD:RCL %d",
+        """Set the battery model from internal memory.""",
+        validator=strict_discrete_set,
+        values=_INTERNAL_MEMORY_SLOTS
+    )
+
+    # Needs two arguments
+    # bs_load_model_usb = Instrument.setting(
+    #    ":BATT:MOD:LOAD:USB %d, \"%s\"",
+    #    """Loads a battery model from USB to the internal memory.""",
+    #    validator=strict_discrete_set,
+    #    values=_INTERNAL_MEMORY_SLOTS
+    #    cast=int
+    # )
```

Comment:
@BenediktBurger thanks for the review so far. Maybe you can give me a hint on this one.  
I would need two arguments here: The internal memory slot where to store the battery profile and the name of the battery-profile on an USB-Stick.  
Is it possible to have two arguments using pymeasure? Otherwise i would hardcode the internal memory slot to be used.

Reply:
In this case, a method (not a property) is better suited.